File: solutions/java/sum-of-multiples/2/src/main/java/SumOfMultiples.java
import java.util.HashSet;
import java.util.Set;

class SumOfMultiples {

    Set<Integer> factors; 
    int num;

    SumOfMultiples(int number, int[] set) {
        factors = new HashSet<Integer>();
        for(int f : set) {
            factors.add(f);
        }
        num =  number;
    }

    int getSum() {
        Set<Integer> s = new HashSet<Integer>();
        int sum=0;
        for(int f : factors){
            for(int m=f; m<num && m!=0; m+=f){
                if(s.add(m)){
                    sum +=m;
                }
            }
        }
        return sum;
    }

}

File: solutions/cpp/sum-of-multiples/5/sum_of_multiples.cpp
#include "sum_of_multiples.h"

#include <set>
#include <numeric>

using namespace std;

namespace sum_of_multiples {

    int to(set<int> factors, int limit){
        int sum{0};
        for(int m=1; m<limit; m++){
            for(int f : factors){
                    if (f!=0 && m % f==0) {
                        sum +=m;
                        break;
                    }
            }
        }
        return sum;
    }
    

}  // namespace sum_of_multiples
